src/crimson/osd/pg_recovery.cc

```diff
@@ -523,8 +523,8 @@ void PGRecovery::on_backfill_reserved()
   using BackfillState = crimson::osd::BackfillState;
   backfill_state = std::make_unique<BackfillState>(
     *this,
-    std::make_unique<BackfillState::PeeringFacade>(pg->get_peering_state()),
-    std::make_unique<BackfillState::PGFacade>(
+    std::make_unique<crimson::osd::PeeringFacade>(pg->get_peering_state()),
+    std::make_unique<crimson::osd::PGFacade>(
```

Comment:
why do we need the cast?

Reply:
`crimson::osd::BackfillState::PGFacade` is the interface, `crimson::osd::PGFacade` is the crimson's implementation of it.